views/create_account.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * PEP Capping 2017 Algozzine's Class
+ *
+ * Displays a page to create a new account.
+ *
+ * This page is accessed when a new LDAP account does
+ * not have a corresponding entry in the employees table.
+ * This will create an employee account with a default
+ * role of "NewUser".
+ *
+ * @author Jack Grzechowiak
+ * @copyright 2017 Marist College
+ * @version 0.2
+ * @since 0.1
+ */
+
+global $params, $db;
+include ('../models/Notification.php');
+
+// Make sure LDAP credentials are present
+if (!isset($_SESSION['ldaprdn']))
+    header('Location: /login');
+
+$error = "";
+
+if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+    $email = $_SESSION['ldaprdn'];
+    $fname = valueOrEmpty($_POST, 'firstName');
+    $mInit = valueOrEmpty($_POST, 'middleInit');
+    $lname = valueOrEmpty($_POST, 'lastName');
+    $phone = valueOrEmpty($_POST, 'primaryPhone');
+
+    // Create new employee / person
+    $res = $db->query("SELECT employeeinsert(fname := $1::TEXT, lname := $2::TEXT, ".
+        "mInit := $3::VARCHAR, em := $4::TEXT, pPhone := $5::TEXT, pLevel := 'New'::PERMISSION)",
+        [$fname, $lname, $mInit, $email, $phone]);
+    if ($res) {
+
+        // Check for query errors
+        $state = pg_result_error_field($res, PGSQL_DIAG_SQLSTATE);
+        pg_free_result($res);
+        if ($state == 0) {
+
+            // Get new employee information to login
+            $res = $db->query("SELECT firstname, lastname, permissionlevel, employeeid FROM people, employees WHERE employees.email = $1 ".
+                "AND employees.employeeid = people.peopleid", [$email]);
+            $state = pg_result_error_field($res, PGSQL_DIAG_SQLSTATE);
+            if ($state == 0 && pg_num_rows($res) > 0) {
+
+                // Set information in session (simulates login)
+                $info = pg_fetch_assoc($res);
+                pg_free_result($res);
+                $_SESSION['employeeid'] = $info['employeeid'];
+                $_SESSION['username'] = $info['firstname'] . ' ' . $info['lastname'];
+                $_SESSION['role'] = Role::roleFromPermissionLevel($info['permissionlevel']);
+                header('Location: /dashboard');
+                die();
+            } else {
+                pg_free_result($res);
+                $error = $state;
+            }
+        } else {
+            $error = $state;
+        }
+    } else {
+        $error = "-1";
```

Comment:
This error should probably be more specific. It doesn't like like it _should_ ever happen, all the more reason to self document.